### PR TITLE
Move symfony dependencies to "suggest"

### PR DESCRIPTION
**Pull request by oligriffiths**
The symfony dependencies are only requires for CLI usage.
Move them to "suggest" so the library can be used without depending upon them.

Per https://github.com/ua-parser/uap-php/issues/43

Better still would be to make a new `cli` package that depends upon `uap-php` and remove symfony from this package entirely

### CODE REVIEW

**Comment by staabm** [comment]
as the dependency is optional now, you should add a `class_exists()` or similar check into the parts which require symfony-classes and suggest to the user to add those dependencies.

otherwise the DX will be horrible after merging this

File: composer.json [hunk 1]
```diff
@@ -2,15 +2,17 @@
...
-        "symfony/filesystem": "^2.0 || ^3.0 || ^4.0",
-        "symfony/finder": "^2.0 || ^3.0 || ^4.0",
-        "symfony/console": "^2.0 || ^3.0 || ^4.0",
         "php": ">=5.3.0"
     },
     "require-dev": {
         "phpunit/phpunit": "^4.0 || ^5.0"
     },
+    "suggest": {
+        "symfony/yaml": "Required for CLI usage",
```
Comment (staabm) [inline composer.json]: maybe mention minimum version?

Comment (oligriffiths) [reply composer.json]: Of 2.0?

File: bin/uaparser [hunk 2]
```diff
@@ -2,6 +2,12 @@
 <?php
 namespace UAParser\Command;
 
+// Ensure symfony packages installed
+if (!class_exists(Symfony\Component\Console\Application::class)) {
```
Comment (staabm) [inline bin/uaparser]: this will only work after autoloading was included, I guess.

Comment (oligriffiths) [reply bin/uaparser]: Na, php will convert that to a string regardless of autoloading. Allows for cleaner refactoring and usage finders though.

Comment (staabm) [reply bin/uaparser]: I mean it will never find this symfony classes as the composer autoloading is required a few lines below

Comment (oligriffiths) [reply bin/uaparser]: `php -a`
```
php > var_dump(Foo\Bar::class);
string(7) "Foo\Bar"
```
The autoloader isn't required, PHP just interpolates that syntax into a string. magic 😄

Comment (staabm) [reply bin/uaparser]: But it cannot find the class when composer is not required

**Comment by oligriffiths** [comment]
FYI `composer install --install-suggests` is required in travis for tests to pass.

File: bin/uaparser (hunk 2 above)
Comment (oligriffiths) [reply bin/uaparser]: I promise you that isn't the case. PHP doesn't invoke the autoloader when doing `::class` on the end of a class. Exactly why running that snipped in an interactive shell with no autoloader registered doesn't cause a `class not found` error

**Comment by rpkamp** [comment]
I would install it as a dev-dependency since you want the dev install to be as complete as possible.

**Comment by rpkamp** [comment]
I'm sorry I was a bit vague. I meant require-dev as well as suggest. Require-dev so the people working on it don't need to anything special to get the tests to run, and suggest for the end users.

**Comment by oligriffiths** [comment]
Ah, makes sense. Will update

File: bin/uaparser (hunk 2 above)
Comment (rpkamp) [reply bin/uaparser]: Nothing better than a small experiment to show and tell :smiley: 

```
<?php
namespace UAParser\Command;

echo Symfony\Component\Console\Application::class .
    (class_exists(Symfony\Component\Console\Application::class) ? ' exists' : ' does not exist'), PHP_EOL;

require_once(__DIR__.'/../vendor/autoload.php');

echo Symfony\Component\Console\Application::class .
    (class_exists(Symfony\Component\Console\Application::class) ? ' exists' : ' does not exist'), PHP_EOL;
```

The output may surprise you, it is:

```
UAParser\Command\Symfony\Component\Console\Application does not exist
UAParser\Command\Symfony\Component\Console\Application does not exist
```

The problem is that the `Symfony\Component\Console\Application` class needs to be prefixed with a \ to indicate it lives in the root namespace and is not a part of this namespace (the `use` statement later on does not change this). With that fixed, let's try again:

```php
<?php
namespace UAParser\Command;

echo \Symfony\Component\Console\Application::class .
    (class_exists(\Symfony\Component\Console\Application::class) ? ' exists' : ' does not exist'), PHP_EOL;

require_once(__DIR__.'/../vendor/autoload.php');

echo \Symfony\Component\Console\Application::class .
    (class_exists(\Symfony\Component\Console\Application::class) ? ' exists' : ' does not exist'), PHP_EOL;
```

and the output is now:

```
Symfony\Component\Console\Application does not exist
Symfony\Component\Console\Application exists
```

So basically:

1. The check needs to be performed _after_ the autoloader has been loaded
2. The check needs to be for `\Symfony\Component\Console\Application::class` instead of `Symfony\Component\Console\Application::class`

**Comment by oligriffiths** [comment]
Cool. Will update

**Comment by hexus** [comment]
Would be great to see this merged and released! 😁
